### ISSUE TITLE
How to make dropout 'not scaling' during training?

### ISSUE THREAD

**Issue by skynbe** [pytorch/pytorch#7544]
Dropout always scale 1/(1-p) during training, but I want to get the original outputs. How to get it?

**Comment by apaszke** [comment]
You can set the modules to evaluation mode, even though you will still be training.

**Comment by Noiredd** [comment]
...except in evaluation mode dropout does nothing. I don't think this is what the OP wanted.

It would be pretty useful to be able to have some flag, e.g. `torch.nn.Dropout(p=0.5, inplace=False, scaling=True)`, so that we could use only the random-zeroing functionality. But I see this is hardcoded this way in the CUDA kernels for dropout... :/

**Comment by mach881040** [comment]
bump

**Comment by andreimargeloiu** [comment]
thumbs up

**Comment by p0p4k** [comment]
Just multiply the output of dropout layer with (1-p) ?

**Comment by jmercat** [comment]
```
class InputDropout(nn.Module):
    """
        Dropout without rescaling.
    """
    def __init__(self, rate) -> None:
        super().__init__()
        self.rate = rate
        
    def forward(self, x: torch.Tensor) -> torch.Tensor:
        mask = torch.empty(x.shape, device=x.device).bernoulli_(1-self.rate)
        return x * mask

```

This would achieve it
